shopfloor/actions/data.py

```diff
@@ -243,9 +271,27 @@ def _product_supplier_code(self, rec, field):
     @ensure_model("stock.picking.batch")
     def picking_batch(self, record, with_pickings=False, **kw):
         parser = self._picking_batch_parser
-        if with_pickings:
+        if with_pickings is True:
             parser.append(("picking_ids:pickings", self._picking_parser))
-        return self._jsonify(record, parser, **kw)
+        if with_pickings == "full":
+            parser.append(("picking_ids:pickings", self._full_picking_parser))
+
+        data = self._jsonify(record, parser, **kw)
+
+        if with_pickings == "full":
+            pickings = record.picking_ids
+
+            for i, picking in enumerate(pickings):
```

Comment:
IMO this should be done in the service where you need it.
Anyway, these move lines computation shall be done in the full_picking_parser.